crates/turbopack-node/src/evaluate.rs

```diff
@@ -242,80 +254,133 @@ pub async fn evaluate(
         runtime_entries,
         additional_invalidation,
         debug,
-    )
-    .await?;
-
-    let args = args.into_iter().try_join().await?;
-    // Assume this is a one-off operation, so we can kill the process
-    // TODO use a better way to decide that.
-    let kill = args.is_empty();
-
-    // Workers in the pool could be in a bad state that we didn't detect yet.
-    // The bad state might even be unnoticable until we actually send the job to the
-    // worker. So we retry picking workers from the pools until we succeed
-    // sending the job.
-
-    let (mut operation, _) = FutureRetry::new(
-        || async {
-            let mut operation = pool.operation().await?;
-            operation
-                .send(EvalJavaScriptOutgoingMessage::Evaluate {
-                    args: args.iter().map(|v| &**v).collect(),
-                })
-                .await?;
-            Ok(operation)
-        },
-        PoolErrorHandler,
-    )
-    .await
-    .map_err(|(e, _)| e)?;
+    );
+
+    // Note the following code uses some hacks to create a child task that produces
+    // a stream that is returned by this task.
 
-    // The evaluation sent an initial intermediate value without completing. We'll
-    // need to spawn a new thread to continually pull data out of the process,
-    // and ferry that along.
-    let stream = JavaScriptStream::new_open(
+    // We create a new cell in this task, which will be updated from the
+    // [compute_evaluate_stream] task.
+    let cell = turbo_tasks::macro_helpers::find_cell_by_type(*JAVASCRIPTEVALUATION_VALUE_TYPE_ID);
```

Comment:
This needs to go into a `turbo-tasks` primitive. We shouldn't expose cell APIs or private macro generated identifiers outside of turbo tasks.